src/Sylius/Behat/Context/Ui/Shop/ContactContext.php

```diff
@@ -0,0 +1,141 @@
+     */
+    public function __construct(
+        ContactPageInterface $contactPage,
+        NotificationCheckerInterface $notificationChecker
+    ) {
+        $this->contactPage = $contactPage;
+        $this->notificationChecker = $notificationChecker;
+    }
+
+    /**
+     * @When I want to request contact
+     */
+    public function iWantToRequestContact()
+    {
+        $this->contactPage->open();
+    }
+
+    /**
+     * @When I specify the email as :email
+     * @When I do not specify the email
+     */
+    public function iSpecifyTheEmail($email = null)
+    {
+        $this->contactPage->specifyEmail($email);
+    }
+
+    /**
+     * @When I specify the message as :message
+     * @When I do not specify the message
+     */
+    public function iSpecifyTheMessage($message = null)
+    {
+        $this->contactPage->specifyMessage($message);
+    }
+
+    /**
+     * @When I send it
+     * @When I try to send it
+     */
+    public function iSendIt()
+    {
+        $this->contactPage->send();
+    }
+
+    /**
+     * @Then I should be notified that the contact request has been submitted successfully
+     */
+    public function iShouldBeNotifiedThatTheContactRequestHasBeenSubmittedSuccessfully()
+    {
+        $this->notificationChecker->checkNotification(
+            'Your contact request has been submitted successfully.',
+            NotificationType::success()
+        );
+    }
+
+    /**
+     * @Then /^I should be notified that the (email|message) is required$/
+     */
+    public function iShouldBeNotifiedThatElementIsRequired($element)
+    {
+        $this->assertFieldValidationMessage(
+            $this->contactPage,
+            $element,
+            sprintf('Please enter your %s.', $element)
+        );
+    }
+
+    /**
+     * @Then /^I should be notified that the (email) is invalid$/
+     */
+    public function iShouldBeNotifiedThatElementIsInvalid($element)
+    {
+        $this->assertFieldValidationMessage(
+            $this->contactPage,
+            $element,
+            sprintf('This %s is invalid.', $element)
+        );
+    }
+
+    /**
+     * @Then /^I should be notified that there was a problem with sending a contact request$/
+     */
+    public function iShouldBeNotifiedThatThereWasAProblemWithSendingAContactRequest()
+    {
+        $this->notificationChecker->checkNotification(
+            'There was a problem with sending a contact request. Please try again later.',
+            NotificationType::failure()
+        );
+    }
+
+    /**
+     * @param PageInterface $page
+     * @param string $element
+     * @param string $expectedMessage
+     */
+    private function assertFieldValidationMessage(PageInterface $page, $element, $expectedMessage)
+    {
+        Assert::true(
+            $page->checkValidationMessageFor($element, $expectedMessage),
+            sprintf('There should be a message: "%s".', $expectedMessage)
```

Comment:
👍 
When you have a page method that can return something meaningful, do it and deal with assertion in context, rather then `x === y` in the page itself.
Doing too much of logic in pages cripples exception messages, and prolongs debugging for future devs